src/tasks/trainerUtils.py

```diff
@@ -158,115 +164,109 @@ def generate_trainer_name(raw_trainer_name, pokemon1_level):
+        temp_celebi_IDs = parse_randomized_teams(file_path, "", team_num, lookup)
+        add_trainers(zoneID, temp_celebi_IDs, [lookup] * len(temp_celebi_IDs))
+        return trainers
+        
+    else:
+        if BARRY in lookup:
+            temp_trainer_IDs = parse_randomized_teams(file_path, lookup, team_num, None)
+            for ID in temp_trainer_IDs:
+                team_type = lookup.split("_")[-1].strip('"')
+                trainer = get_single_trainer(zoneID, ID, temp_trainer_IDs, team_type)
+                trainers.append(trainer)
+            return trainers
+        temp_trainer_IDs = parse_randomized_teams(file_path, lookup, team_num, None)
+        add_trainers(zoneID, temp_trainer_IDs)
+        for ID in temp_trainer_IDs:
+            trainer = get_single_trainer(zoneID, ID, temp_trainer_IDs, None)
+            trainers.append(trainer)
+        return trainers
 
-    with open(file_path, 'r', encoding="utf-8") as f:
-        for line in f:
-            substrings = line.split('\n')
-            for substring in substrings:
-                areaName = file_path.split("/")[-1].split(".")[0].upper()
-                zoneID = get_zoneID(areaName)
-                gym_leader_lookup = f"ev_{areaName.lower()}_randomteam"
-                
-                if TRAINER_BATTLE in substring or MULTI_TRAINER_BATTLE in substring:
-                    args = parse_trainer_btl_set(substring.strip())
-                else:
-                    continue
+def get_assorted_trainer_data(file_path, areaName, zoneID, trainerID1, trainerID2, args):
+    count_keeper = []
+    trainers = []
+    cyrus_lookup = f"ev_{areaName.lower()}_randomteam_cyrus"
+    for starter in STARTER:
+        rival_lookup = f"ev_{areaName.lower()}_randomteam_barry_{starter}"
+        rival_teams = get_random_team_data(file_path, areaName, zoneID, trainerID1, trainerID2, rival_lookup, 4)
+        trainers.extend(rival_teams)
+    if rival_teams:
+        return trainers
+    if count_keeper == []:
+        cyrus_teams = get_random_team_data(file_path, areaName, zoneID, trainerID1, trainerID2, cyrus_lookup, 4)
+        if cyrus_teams:
+            trainers.extend(cyrus_teams)
+            return trainers
+    if count_keeper == []:
+        master_teams = get_random_team_data(file_path, areaName, zoneID, trainerID1, trainerID2, MASTER_TRAINER, MASTER_TRAINER_TYPES)
+        if master_teams:
+            trainers.extend(master_teams)
+            return trainers
+    if count_keeper == []:
+        celebi_teams = get_random_team_data(file_path, areaName, zoneID, trainerID1, trainerID2, CELEBI, 7)
+        if celebi_teams:
+            trainers.extend(celebi_teams)
+            return trainers
+    if count_keeper == []:
+        print("Lucas and Dawn's Single Battles are not yet supported:", areaName, args)
+        return trainers
+        '''
+        ### This section is currently in the progress of being improved and finalized in ev_script.
```

Comment:
Remove unused code

Reply:
This is mostly for when the ev script is finalized but I guess I have a reminder to get it done every time the script runs.